frontend/src/templates/Field/Date/DateField.tsx

```diff
@@ -0,0 +1,69 @@
+import { useCallback, useMemo } from 'react'
+import { Controller } from 'react-hook-form'
+import { isAfter, isBefore, startOfToday } from 'date-fns'
+
+import {
+  DateFieldBase,
+  DateSelectedValidation,
+  FormFieldWithId,
+} from '~shared/types/field'
+
+import { createDateValidationRules } from '~utils/fieldValidation'
+import DateInput from '~components/DatePicker'
+
+import { BaseFieldProps, FieldContainer } from '../FieldContainer'
+
+export type DateFieldSchema = FormFieldWithId<DateFieldBase>
+export interface DateFieldProps extends BaseFieldProps {
+  schema: DateFieldSchema
+}
+
+/**
+ * @precondition Must have a parent `react-hook-form#FormProvider` component.
+ */
+export const DateField = ({
+  schema,
+  questionNumber,
+}: DateFieldProps): JSX.Element => {
+  const validationRules = useMemo(
+    () => createDateValidationRules(schema),
+    [schema],
+  )
+
+  const isDateUnavailable = useCallback(
+    (date: Date) => {
+      const { selectedDateValidation } = schema.dateValidation
+      // All dates available.
+      if (!selectedDateValidation) return false
+
+      switch (selectedDateValidation) {
+        case DateSelectedValidation.NoPast:
+          return isBefore(date, startOfToday())
+        case DateSelectedValidation.NoFuture:
+          return isBefore(startOfToday(), date)
```

Comment:
what if somehow the date given is during the day today? then today's date would be made unavailable, when actually it should be available.

would `isAfter(endOfToday(), date)` be more accurate?

Reply:
You're right; even though the date that is given is always only in the ISO `YYYY-MM-DD` format, its better to be safe than sorry. Will update.

Reply:
Changed it to `isAfter(date, endOfToday())`